snorkel/contrib/snark/annotations.py

```diff
@@ -0,0 +1,147 @@
+from functools import partial
+from pickle import loads
+
+from models.candidate import Candidate
+from models.context import Sentence, Span
+from snorkel.annotations import load_label_matrix
+from snorkel.models.annotation import Label, LabelKey
+from snorkel.models.meta import snorkel_conn_string
+from snorkel.models.views import create_serialized_candidate_view
+
+
+class SparkLabelAnnotator:
+    """
+    Distributes candidates to a Spark cluster and applies labeling functions over them.
+
+    Distributed candidate sets are cached by split for the annotator, so repeated calls to apply() will
+    operate over the same split of candidates as the first call for that split, regardless of how the
+    underlying database changes. To reload a candidate set split, construct a new SparkLabelAnnotator.
```

Comment:
This could be pretty bad failure mode if user doesn't read doc string... should print warning or something?